### PR TITLE
Code Style Fixes (Rubocop)

### PR DESCRIPTION
**Pull request by mheap**
## Description

Apply code style auto fixes, plus some changes that it couldn't make itself (e.g. guard clauses rather than wrapping a block in an `if` statement. Things look ok at first glance, but I'd like to test on a review app and run an automated checker to see if any pages now `500`

## Deploy Notes

There are a *lot* of changes, so please be extra-careful when reviewing this. You'll probably want to test on a review app rather than reading the diff.

### CODE REVIEW

**Comment by mheap** [comment]
I've played with the review app and tested the following:

* Feedback positive/negative still works
* Slack invitation still works
* OAS renders
* Markdown renders
* Building blocks render
* Legacy API docs render

I'm going to remove the `don't merge` label, but would appreciate someone else trying this out too

File: app/filters/code_filter.rb [hunk 1]
```diff
@@ -23,12 +23,6 @@ def call(input)
 
       highlighted_source = highlight(code, lexer)
 
-      line_numbers = (1..total_lines).map do |line_number|
```
Comment (ChrisGuzman) [inline app/filters/code_filter.rb]: Was this deleted by rubocop or was done manually? Is it something we need?

File: app/views/layouts/partials/_footer.html.erb [hunk 2]
```diff
@@ -53,13 +53,6 @@
         <span class="api-status-status">Loading</span>
       </a>
 
-      <div id="footer-social-links">
```
Comment (ChrisGuzman) [inline app/views/layouts/partials/_footer.html.erb]: Why get rid of this?

File: app/filters/code_filter.rb (hunk 1 above)
Comment (mheap) [reply app/filters/code_filter.rb]: Manually, it's unused

File: app/views/layouts/partials/_footer.html.erb (hunk 2 above)
Comment (mheap) [reply app/views/layouts/partials/_footer.html.erb]: This was a mistake, I'll put it back

**Pull request merged**